pkg/ratelimit/runner.go

```diff
@@ -144,16 +158,25 @@ func (cr *ConcurrentRunner) run(ctx context.Context, task func(context.Context),
 func (cr *ConcurrentRunner) processPendingTasks() {
 	cr.pendingMu.Lock()
 	defer cr.pendingMu.Unlock()
-	for len(cr.pendingTasks) > 0 {
-		task := cr.pendingTasks[0]
+	if len(cr.pendingHighPriorityTasks) > 0 {
+		task := cr.pendingHighPriorityTasks[0]
+		select {
+		case cr.taskChan <- task:
```

Comment:
Here, I have some confusion. Even though we use two queues to distinguish priority, we dispatch the tasks one by one, and they run concurrently. It is not very relevant to the priority because we cannot control the priority during the runtime process. The main difference lies in the elimination strategy of pending tasks. So, if we split the runner and let each runner have its own elimination strategy for pending tasks, would it be clearer?

Reply:
What do you mean about "control the priority during the runtime process"?

Reply:
like unified-pool.

Reply:
> Will it lead to normal priority task accumulation 
No, the normal priority task will be restricted by the maximum wait time.

> I think splitting it into two runners and isolating the light task and heavy task is more reasonable.
Split into two runners will also compete for the same resource which may slow down the speed of executing the high priority tasks

Reply:
> Split into two runners will also compete for the same resource which may slow down the speed of executing the high-priority tasks.

why?

Reply:
Even though the task is low priority, it might also acquire the lock.

Reply:
they only share the root tree lock for get region sometime, I think impact won't be significant.

Reply:
The reason why we need to distinguish the priority is that we want these high-priority tasks can be executed ASAP.

Reply:
Split runners won't take effect.

Reply:
Considering that the current implementation method is to process normal tasks only after all high-priority tasks are completed, will this lead to starvation? Do we need to change it to a probabilistic selection method based on task priority before each execution?